src/Parser/Monk/Brewmaster/Modules/Spells/BreathOfFire.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Wrapper from 'common/Wrapper';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+class BreathOfFire extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+
+  get uptime() {
+    return this.enemies.getBuffUptime(SPELLS.BREATH_OF_FIRE_DEBUFF.id) / this.owner.fightDuration;
+  }
```

Comment:
Wouldn't this "break" in any AoE scenario - since you have multiple debuffs up at once?

Reply:
Potentially. I looked at how AffLock debuffs were tracked and then did the same thing. It should probably be changed to the same metric as ISB (fraction of hits mitigated with it) since it may never be applied to adds that don't hit you. It works well enough on most fights in Antorus that it is worth replacing the old one (which completely breaks with the chest since we only want to cast often enough to maintain the buff).